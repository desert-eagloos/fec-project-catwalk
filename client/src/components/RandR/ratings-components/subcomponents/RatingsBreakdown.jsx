import React from 'react';
import { Row, Col, ProgressBar } from 'react-bootstrap';
import PropTypes from 'prop-types';

import '../../../../css/RandR/Ratings/RatingBreakdown.css';

export default function RatingsBreakdown({ ratings }) {
  const totalRatings = Object.values(ratings)
    .reduce((sum, x) => sum + Number(x), 0);

  const rand = () => Math.floor(Math.random() * 1);

  return (
    Object.entries(ratings)
      .sort((a, b) => b[0].localeCompare(a[0]))
      .map(([rating, value], i) => (
        <Row key={`row${i + rand()}`} className="m-0 p-0">
          <Col sm={2} className="m-0 p-0">
            {rating}
          </Col>
          <Col className="m-0 p-0">
            <ProgressBar now={(value / totalRatings) * 100} />
          </Col>
        </Row>
      ))
  );
}

RatingsBreakdown.defaultProps = {
  ratings: {
    5: 1,
    4: 3,
    3: 2,
    2: 1,
    1: 0
  },
};

RatingsBreakdown.propTypes = {
  ratings: PropTypes.shape({
    5: PropTypes.number,
    4: PropTypes.number,
    3: PropTypes.number,
    2: PropTypes.number,
    1: PropTypes.number,
  }),
};
