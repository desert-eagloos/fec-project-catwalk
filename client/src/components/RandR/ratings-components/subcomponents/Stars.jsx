import React from 'react';

export default function Stars(rating) {
  return (
    <div className="stars-outer">
      <div className="stars-inner" style={{ width: rating }} />
    </div>
  );
}
