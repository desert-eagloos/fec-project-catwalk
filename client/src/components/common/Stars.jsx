import React from 'react';
import Rating from 'react-rating';
import PropTypes from 'prop-types';

import { roundToNearestQuarter } from '../../utils/ratings';

import 'font-awesome/css/font-awesome.css';

export default function Stars({ rating }) {
  return (
    <>
      <Rating
        emptySymbol="fa fa-star-o"
        fullSymbol="fa fa-star"
        initialRating={roundToNearestQuarter(rating)}
        fractions={4}
      />
    </>
  );
}

Stars.defaultProps = {
  rating: 3,
};

Stars.propTypes = {
  rating: PropTypes.number,
};
