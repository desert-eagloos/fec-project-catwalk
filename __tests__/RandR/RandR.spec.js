import React from 'react';
import renderer from 'react-test-renderer';
import Stars from '../../client/src/components/RandR/ratings-components/subcomponents/Stars';
import RatingsBreakdown from '../../client/src/components/RandR/ratings-components/subcomponents/RatingsBreakdown';

describe('Creates Ratings and Reviews Components', () => {
  describe('Creates a Star Rating Component', () => {
    test('Creates a Star', () => {
      const component = renderer.create(<Stars />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });
  describe('Creates a Rating Breakdown Component', () => {
    test('Creates a Rating Breakdown', () => {
      const ratings = {
        5: 2,
        4: 3,
        3: 5,
        2: 1,
        1: 10,
      };
      const component = renderer.create(<RatingsBreakdown ratings={ratings} />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });
});