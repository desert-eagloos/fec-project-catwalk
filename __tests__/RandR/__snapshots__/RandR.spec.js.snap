// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates Ratings and Reviews Components Creates a Rating Breakdown Component Creates a Rating Breakdown 1`] = `
<div
  className="breakdown-container"
>
  <div
    className="breakdown-rating"
  >
    <span>
      5
    </span>
    <span>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={9.523809523809524}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "9.524%",
            }
          }
        />
      </div>
    </span>
  </div>
  <div
    className="breakdown-rating"
  >
    <span>
      4
    </span>
    <span>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={14.285714285714285}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "14.286%",
            }
          }
        />
      </div>
    </span>
  </div>
  <div
    className="breakdown-rating"
  >
    <span>
      3
    </span>
    <span>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={23.809523809523807}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "23.81%",
            }
          }
        />
      </div>
    </span>
  </div>
  <div
    className="breakdown-rating"
  >
    <span>
      2
    </span>
    <span>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={4.761904761904762}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "4.762%",
            }
          }
        />
      </div>
    </span>
  </div>
  <div
    className="breakdown-rating"
  >
    <span>
      1
    </span>
    <span>
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={47.61904761904761}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "47.619%",
            }
          }
        />
      </div>
    </span>
  </div>
</div>
`;

exports[`Creates Ratings and Reviews Components Creates a Star Rating Component Creates a Star 1`] = `
<div>
  <span
    onMouseLeave={[Function]}
    style={
      Object {
        "direction": "ltr",
        "display": "inline-block",
      }
    }
  >
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star"
        />
      </span>
    </span>
  </span>
</div>
`;
